sdks/python/apache_beam/io/gcp/tests/bigquery_matcher.py

```diff
@@ -165,6 +169,33 @@ def describe_mismatch(self, pipeline_result, mismatch_description):
       .append_text(self.actual_data)
 
 
+class BigqueryFullResultStreamingMatcher(BigqueryFullResultMatcher):
+  """
+  Matcher that verifies Bigquery data with given query.
+
+  Fetch Bigquery data with given query, compare to the expected data.
+  This matcher polls BigQuery until the no. of records in BigQuery is
+  equal to the no. of records in expected data.
+  A timeout can be specified.
+  """
+
+  def __init__(self, project, query, data, timeout=MAX_TIMESTAMP):
```

Comment:
Could you make the default timeout to be finite. Would 5/10/15 minutes work? (the less the better - having post-commit tests hang until their 2 hour deadline is not fun. It easier to debug where the test timed out if it raises a TimeoutError and the stacktrace is printed)

Reply:
I think 5 minutes should be more than enough. I'll make the change.